CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+cmake_minimum_required(VERSION 3.10)
+
+# Set the project name and version
+project(cli VERSION 1.0 LANGUAGES CXX)
+
+# Set the C++ standard
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
```

Comment:
You build it cross platform, so it is encouraged to compile pure C++ standard without any platform extensions, so consider adding `CMAKE_CXX_EXTENSIONS OFF`. That way your `-std` on Linux will be `c++17` instead of `gnu++17` and so on